kofta/src/app/pages/ViewUserPage.tsx

```diff
@@ -19,22 +20,29 @@ export const ViewUserPage = () => {
     params: { username },
   } = useRouteMatch<{ username: string }>();
 
-  const { data: profileFromDB, isLoading } = useQuery<UserWithFollowInfo>(
-    ["get_user_profile", username],
-    () => {
-      return wsFetch<any>({
-        op: "get_user_profile",
-        d: { userId: username },
-      });
-    },
-    { enabled: !profileFromState && status === "auth-good" && !!username }
+  const isDirectFromUrl = !profileFromState && !!username;
+
+  const { isLoading, data: profileFromDB } = useUserProfileQuery(username , isDirectFromUrl && status === "auth-good" );
```

Comment:
What is happening here? We are fetching the profile info if the user comes in directly/externally from a URL?

Reply:
Yes
First fetch user profile by username, then fetch if current logged in user is following that user